### PR TITLE
pkg/uroot: move /uroot package in; remove unused code.

### PR DESCRIPTION
**Pull request by hugelgupf**
This moves the original "uroot" package under pkg/uroot/util. Also
contains cleanup of the package; and fixing up any invocations.

Signed-off-by: Christopher Koch <chrisko@google.com>

/edit: Argh, spoke too soon. cmds/init uses some of the stuff I removed.

### CODE REVIEW

File: pkg/uroot/util/uname.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// Copyright 2015-2017 the u-root Authors. All rights reserved
```
Comment (rminnich) [inline pkg/uroot/util/uname.go]: I thought all the stuff in x/unix replaced this? I have not looked much however.

Comment (hugelgupf) [reply pkg/uroot/util/uname.go]: x/unix doesn't have a Uname for darwin, so we still need the darwin/linux split.

**Comment by hugelgupf** [comment]
This should build now.

File: pkg/uroot/util/uname.go (hunk 1 above)
Comment (tklauser) [reply pkg/uroot/util/uname.go]: FWIW I just sent https://golang.org/cl/77270 which adds Uname for darwin. Once the CL is approved, x/sys/unix could be re-vendored and would make this wrapper unnecessary.

Comment (bkochendorfer) [reply pkg/uroot/util/uname.go]: Nice @tklauser

File: cmds/mm/mm.go [hunk 2]
```diff
@@ -11,36 +11,28 @@ import (
 	"os/exec"
 	"syscall"
```
Comment (bkochendorfer) [inline cmds/mm/mm.go]: `mm/mm.go:12:2: imported and not used: "syscall"`

**Comment by ryukinix** [comment]
image


@hugelgupf, after fix `syscall` module import error as @bkochendorfer pointed, we got that above.

**Pull request merged**